core/src/main/resources/jenkins/model/Jenkins/_safeRestart.jelly

```diff
@@ -25,16 +25,31 @@ THE SOFTWARE.
 <!-- Delete view -->
 <?jelly escape-by-default='true'?>
 <j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form" xmlns:i="jelly:fmt">
-	<l:layout title="Safely Restart Jenkins">
+	<l:layout title="${%Safe restart}">
 		<st:include page="sidepanel.jelly" />
-		<l:breadcrumb title="${%Safe Restart}" />
+		<l:breadcrumb title="${%Safe restart}" />
 		<l:main-panel>
+			<l:app-bar title="${%Safe restart}" />
 			<j:choose>
 				<j:when test="${app.lifecycle.canRestart()}">
-					<form method="post" action="safeRestart">
-						${%confirmRestart}
-						<f:submit value="${%Yes}" />
-					</form>
+          <p>${%restartWarning}</p>
+
+          <form method="post" action="safeRestart">
+            <f:entry title="${%Jenkins is restarting banner}"
+                     description="${%This will be displayed on the Jenkins dashboard, you can use it to let users know what is happening. A default message will be added if you don't supply one.}">
+              <f:textbox name="message" value="${app.quietDownReason ?: null}"/>
+            </f:entry>
+            <div>
+            <f:submit clazz="jenkins-!-destructive-color" value="${%Restart}"/>
+            </div>
+          </form>
+          <j:if test="${app.isQuietingDown()}">
+            <f:form method="post" action="cancelQuietDown" name="cancelRestart">
+              <p>
+                <f:submit value="${%Cancel}"/>
```

Comment:
Ideally these buttons should be side by side but I haven't managed to make it work because of the different forms, maybe need to combine into one form?

![image](https://github.com/jenkinsci/jenkins/assets/21194782/f28ef468-d5b0-4ae2-b40a-13ae3b8e69b3)

Reply:
`jenkins-!-display-contents` as a class on the form might do the job?

Reply:
doesn't seem to do anything.
I did try using `form` attributes on the input but that breaks form binding here.

Will just need to change the server side code I think

Reply:
<img width="681" alt="image" src="https://github.com/jenkinsci/jenkins/assets/21194782/54adb107-c502-46ec-98ea-5ca1a8d73a49">